object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/CalledMethodsPredicateEvaluator.java

```diff
@@ -15,6 +29,118 @@ public CalledMethodsPredicateEvaluator(final Set<String> cmMethods) {
     this.cmMethods = cmMethods;
   }
 
+  /**
+   * Return true iff the boolean formula "lhs -> rhs" is valid, treating all variables as
+   * uninterpreted.
+   *
+   * <p>Used to determine subtyping between two @CalledMethodsPredicate annotations that are not
```

Comment:
It also handles `@CalledMethodsPredicate` versus `@CalledMethods`.  I suggest just dropping this paragraph, since it's obvious from the code and can easily get out of date.